ttl/ttlworker/job_manager.go

```diff
@@ -0,0 +1,582 @@
+		case <-resizeWorkersTicker:
+			err := m.resizeScanWorkers(int(ScanWorkersCount.Load()))
+			if err != nil {
+				logutil.Logger(m.ctx).Warn("fail to resize scan workers", zap.Error(err))
+			}
+			err = m.resizeDelWorkers(int(DeleteWorkerCount.Load()))
+			if err != nil {
+				logutil.Logger(m.ctx).Warn("fail to resize delete workers", zap.Error(err))
+			}
+		case <-scheduleTicker:
+			m.rescheduleJobs(se, now)
+		}
+	}
+}
+
+func (m *JobManager) resizeScanWorkers(count int) error {
+	var err error
+	m.scanWorkers, err = m.resizeWorkers(m.scanWorkers, count, func() worker {
+		return newScanWorker(m.delCh, m.notifyStateCh, m.sessPool)
+	})
+	return err
+}
+
+func (m *JobManager) resizeDelWorkers(count int) error {
+	var err error
+	m.delWorkers, err = m.resizeWorkers(m.delWorkers, count, func() worker {
+		return newDeleteWorker(m.delCh, m.sessPool)
+	})
+	return err
+}
+
+func (m *JobManager) resizeWorkers(workers []worker, count int, factory func() worker) ([]worker, error) {
+	if count < len(workers) {
+		logutil.Logger(m.ctx).Info("shrink ttl worker", zap.Int("originalCount", len(workers)), zap.Int("newCount", count))
+
+		for _, w := range workers[count:] {
+			w.Stop()
+		}
+		var errs error
+		for _, w := range workers[count:] {
+			err := w.WaitStopped(m.ctx, 30*time.Second)
+			if err != nil {
+				logutil.Logger(m.ctx).Warn("fail to stop ttl worker", zap.Error(err))
+				errs = multierr.Append(errs, err)
+			}
+		}
+
+		// remove the existing workers, and keep the left workers
+		workers = workers[:count]
+		return workers, errs
+	}
+
+	if count > len(workers) {
+		logutil.Logger(m.ctx).Info("scale ttl worker", zap.Int("originalCount", len(workers)), zap.Int("newCount", count))
+
+		for i := len(workers); i < count; i++ {
+			w := factory()
+			w.Start()
+			workers = append(workers, w)
+		}
+		return workers, nil
+	}
+
+	return workers, nil
+}
+
+func (m *JobManager) updateTaskState() {
+	results := m.pollScanWorkerResults()
+	for _, result := range results {
+		job := findJobWithTableID(m.runningJobs, result.task.tbl.ID)
+		if job != nil {
+			logutil.Logger(m.ctx).Debug("scan task state changed", zap.String("jobID", job.id))
+
+			job.finishedScanTaskCounter += 1
+			job.scanTaskErr = multierr.Append(job.scanTaskErr, result.err)
+		}
+	}
+}
+
+func (m *JobManager) pollScanWorkerResults() []*ttlScanTaskExecResult {
+	results := []*ttlScanTaskExecResult{}
+	for _, w := range m.scanWorkers {
+		worker := w.(*ttlScanWorker)
+		result := worker.PollTaskResult()
+		if result != nil {
+			results = append(results, result)
+		}
+	}
+
+	return results
+}
+
+// checkNotOwnJob removes the job whose current job owner is not yourself
+func (m *JobManager) checkNotOwnJob(se session.Session) {
+	for _, job := range m.runningJobs {
+		timeoutJobCtx, cancel := context.WithTimeout(m.ctx, ttlInternalSQLTimeout)
+
+		tableStatus := m.tableStatusCache.Tables[job.tbl.ID]
+		if tableStatus == nil || tableStatus.CurrentJobOwnerID != m.id {
```

Comment:
`tableStatus.CurrentJobOwnerID != m.id` means this job is take over by other worker, so is it necessary to reset the job's status in this node?

Reply:
> reset the job's status in this node

Remove it from the running job and cancelling all running tasks are the only "status" local . The manager cannot get the job anymore :thinking:, so I don't think other states matter.